main.py

```
@@ -0,0 +1,1832 @@
+        
+        # Hardware-Info für Optimierung
+        self.cpu_count = psutil.cpu_count()
+        self.memory_total = psutil.virtual_memory().total
+        self.gpu_available = self._check_gpu_support()
+        
+    def _check_gpu_support(self) -> bool:
+        """Prüft GPU-Verfügbarkeit für Hardware-Beschleunigung."""
+        try:
+            import torch
+            return torch.cuda.is_available()
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)